### PR TITLE
Extra for intro course: tuning

### PR DESCRIPTION
**Pull request by hfrick**
This is a lightweight intro to the very basic concept of "try a bunch of parameters and use the best-performing one". 

We already cover `fit_resamples()` so the step to `tune_grid()` is not very far. Obviously, there is a lot more to tuning but this should get the basic idea across and prepare learners for more sophisticated approaches, either in the advanced course or via e.g. TMwR.

### CODE REVIEW

File: slides/intro-extra-tune.qmd [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+## Specifying tuning parameters
+
+Let's take our previous random forest workflow and tag the number of trees for tuning:
+
+```{r}
+#| label: tag-for-tuning
+#| code-line-numbers: "1|"
+
+rf_spec <- rand_forest(trees = tune()) %>%
```
Comment (simonpcouch) [inline slides/intro-extra-tune.qmd:84]: This is definitely the most intuitive model argument to `rand_forest()`, but I'm a bit [hesitant](https://stats.stackexchange.com/questions/348245/do-we-have-to-tune-the-number-of-trees-in-a-random-forest) to demonstrate tuning with it. It also looks like it doesn't result in marked changes in performance in this example. Would we be down for tuning `mtry` or `min_n` instead?

File: slides/intro-extra-tune.qmd [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+rf_wflow
+```
+
+## Try out multiple values
+
+`tune_grid()` works similar to `fit_resamples()` but covers multiple parameter values:
+
+```{r}
+#| label: rf-tune_grid
+#| code-line-numbers: "2|3-4|5|"
```
Comment (simonpcouch) [inline slides/intro-extra-tune.qmd:97]: Love the use of highlighting in these slides.

File: slides/intro-extra-tune.qmd [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+```{r}
+#| label: rf-results
+
+show_best(rf_res)
+
+best_parameter <- select_best(rf_res)
+best_parameter
+```
+
+`collect_metric()` and `autoplot()` are also available for tune results.
```
Comment (simonpcouch) [inline slides/intro-extra-tune.qmd:120]:
```suggestion
`collect_metrics()` and `autoplot()` are also available for tune results.
```

File: slides/intro-extra-tune.qmd [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+rf_res <- tune_grid(
+  rf_wflow,
+  taxi_folds,
+  grid = 3
+)
+```
+
+## Compare results
+
+Inspecting results and select the best-performing hyperparameter(s):
```
Comment (simonpcouch) [inline slides/intro-extra-tune.qmd:109]:
```suggestion
Inspecting results and selecting the best-performing hyperparameter(s):
```

**Review by simonpcouch — COMMENTED** [review]
I love it!

Once we [decide where it will go](https://github.com/tidymodels/workshops/pull/238#discussion_r1277705924), reminder to drop an entry in `index.qmd`. :)

File: slides/intro-extra-tune.qmd (hunk 1 above)
Comment (hfrick) [reply slides/intro-extra-tune.qmd:84]: `mtry()` would/could mean talking about finalizing parameters so rather `min_n`?

File: slides/intro-extra-tune.qmd (hunk 3 above)
Comment (hfrick) [reply slides/intro-extra-tune.qmd]: thanks for catching that!

File: slides/intro-extra-tune.qmd (hunk 1 above)
Comment (simonpcouch) [reply slides/intro-extra-tune.qmd:84]: Totally!

Comment (hfrick) [reply slides/intro-extra-tune.qmd]: done! [42b756b](https://github.com/tidymodels/workshops/pull/239/commits/42b756b33190be2eb6da6551d1a2ef7b0ed4d819)

**Review by simonpcouch — APPROVED** [review]
:)

**Review by EmilHvitfeldt — APPROVED** [review]
love it!

**Comment by hfrick** [comment]
leaving a note here for 2 missing commits (maybe they are only missing in the web interface?) to follow up on that when the dust has settled:
- updating `index.qmd`
- adding an exercise

**Pull request merged**